docs/src/research/installation.md

```diff
@@ -0,0 +1,60 @@
+
+### Modular Interactive
+
+[Ghaf introduced a modular structure for interactive installer](https://tiiuae.github.io/ghaf/ref_impl/installer.html). The initial Ghaf reference installer still uses raw disk image per target device. In practice, just writes the raw disk image to the target device internal persitent memory - NVMe for example. The key idea with the modular interactive Ghaf installer is to enable customization of the installer per target device needs and at the same time support further development of Ghaf graphical installer. The challenge of the interactive installer is to amount of configurable options, development and testing them. Given the Ghaf approach of target device specific installation described in the last section of [Ghaf Initial Approach](installation.md#ghaf-initial-approach), the requirement for Ghaf device specific installer is challenging. Ghaf installer would have to either:
+
+* embed the device specific installation raw disk image in the installer (current way) which results in huge installer image
+* dynamically build the device specific installation according to the user interactive selection
+* download a pre-built device specific raw disk image which could result in huge number of configurations
+* use some combination of generic and specific (kind of a compromise)
+
+None of which seem feasible in the long run. None of which are either Ghaf objective in the long run.
+
+But how to achieve a device specific secure system installation without getting lost with the generic Linux distro requirements?
+
+### Declarative, Non-Interactive Installation
+
+Now that we already version control reference device specific secure system declarations, the question is if we can transfer those into the device installations without asking user too many questions that make the installation unnecessarily difficult to implement? This alone is not a novel idea. Automatic Original Equipment Manufacturer (OEM) installers have been doing this for long. Those are often not declarative but rather scripted guidance to answer questions in generic installers.
+
+In manual installation, the target device specific disk partitioning has been left to the user. Traditionally in generic installers it is also risk management. A user typically might not want her device disk getting wiped out no questions asked. Of course we could let the user know what is about to happen and ask the user for agreement in confirmation before we install fully automatically. Declarative configurations can handle user preferences. If one wants to change something, it can be changed in the declarations, stored and shared via version control. [Also including the declarative partitioning](https://github.com/nix-community/disko) that has been tested from within the Ghaf installer.
```

Comment:
The target device-specific disk partitioning has been left to the user in manual installation. Traditionally in generic installers, it is also risk management. A user typically might not want her device disk wiped out without questions asked. Of course, we could let the user know what is about to happen and ask the user for agreement in confirmation before we install it fully automatically. Declarative configurations can handle user preferences. If one wants to change something, it can be changed in the declarations, stored, and shared via version control. [Also including the declarative partitioning](https://github.com/nix-community/disko) tested from within the Ghaf installer.

Reply:
Done